samples/isprime/isprime.neon

```diff
@@ -0,0 +1,233 @@
+VAR bQuiet:         Boolean     := FALSE
+
+
+FUNCTION GetRandomA(n:Number): Number
+    RETURN (random.uint32() MOD (n - 2)) + 2
+END FUNCTION
+
+FUNCTION RabinMiller(n: Number, iIterations: Number): Boolean
+    VAR m:       Number := n - 1
+    VAR k:       Number := 0
+
+    IF n < 3 OR NOT odd(n) THEN
+        % Rule: The number n is an EVEN number, and therefore CANNOT be prime.  EXCEPT for the number 2, the ONLY even prime number.
+        IF n = 2 THEN
+            % Since 2 is the ONLY even number that is prime, we handle this case a little differently.
+            RETURN TRUE
+        END IF
+        RETURN FALSE
+    END IF
+
+    IF m > 0 THEN
+        WHILE odd(m) DO
+            INC k
+            m := binary.shiftRight32(m, 1)
+        END WHILE
+    END IF
+
+    VAR r: Number
+    FOR iCount := 1 TO iIterations LABEL WitnessLoop DO
+        LET a: Number := GetRandomA(n)
+        r := 1
+        FOR x := 1 TO m DO
+            r := ((r * a) MOD n)
+        END FOR
+        IF r = 1 OR r = (n - 1) THEN
+            NEXT WitnessLoop
+        END IF
+        FOR x := 0 TO k DO
+            r := ((r*r) MOD n)
+            IF r = 1 THEN
+                RETURN FALSE
+            END IF
+            IF r = (n - 1) THEN
+                NEXT WitnessLoop
+            END IF
+        END FOR
+        RETURN FALSE
+    END FOR
+    RETURN TRUE
+END FUNCTION
+
+FUNCTION HasSmallFactor(n: Number): Boolean
+    FOREACH p IN SMALL_PRIMES DO
+        IF n = p THEN
+            EXIT FOREACH
+        END IF
+        IF (n MOD p = 0) THEN
+            RETURN TRUE
+        END IF
+    END FOREACH
+    RETURN FALSE
+END FUNCTION
+
+FUNCTION output(s: String)
+    IF NOT bQuiet THEN
+        print(s)
+    END IF
+END FUNCTION
+
+FUNCTION Usage()
+    print("")
+    print("Usage: \(sys.args[0]) {-q | -k [num]} {Number | s-e | s- | -e}")
+    print(@@"
+  Where:
+        -k  [num]   Set the number of test iterations.  Values between 2-7 are 
+                    best suited for speed and accuracy.  Default: 5.
+        -q          Quiet output.  Will only output the result of the test.
+    and:
+        s is starting integer, and e is ending integer.
+"@@)
+    sys.exit(1)
+END FUNCTION
+
+BEGIN MAIN
+    IF sys.args.size() <= 1 THEN
+        Usage()
+    END IF
+
+    VAR suppliedNumTests: Boolean := FALSE
+    VAR sval: Number := 0
+    VAR eval: Number := 0
+    VAR nums: Array<String> := []
+    VAR bPar: Boolean := FALSE
+    
+    FOREACH p IN sys.args INDEX i DO
+        IF i = 0 THEN
+            % Skip the first parameter
+            NEXT FOREACH
+        END IF
+        IF bPar THEN
+            bPar := NOT bPar
+            NEXT FOREACH
+        END IF
+        IF p = "-q" OR p = "-Q" THEN
+            bQuiet := TRUE
+        ELSIF p = "-k" OR p = "-K" THEN
+            suppliedNumTests := TRUE
+            bPar := suppliedNumTests
+            TestIterations := num(sys.args[i+1])
```

Comment:
I don't think num() should raise an exception for non-numerical input.  There may be times when you WANT to convert a string to a number, and NOT have it exception on you if there are non-convertible chars in the string.  However, I'm of the opinion that a string of: 123x45 when passed through num(), should return 123, rather than 0.